src/test/fuzz/simplicity.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <primitives/transaction.h>
+extern "C" {
+#include <simplicity/elements/env.h>
+#include <simplicity/elements/exec.h>
+}
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <vector>
+
+FUZZ_TARGET(simplicity)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    {
+        const std::optional<CMutableTransaction> mtx_precomputed = ConsumeDeserializable<CMutableTransaction>(fuzzed_data_provider);
+        if (mtx_precomputed) {
+            const CTransaction tx_precomputed{*mtx_precomputed};
+            const PrecomputedTransactionData precomputed_transaction_data{tx_precomputed};
+            const transaction* tx = precomputed_transaction_data.m_simplicity_tx_data;
+
+            const uint256 genesisBlockHash = precomputed_transaction_data.m_hash_genesis_block;
+
+            std::vector<unsigned char> imr = fuzzed_data_provider.ConsumeBytes<unsigned char>(32);
+
+            const uint_fast32_t ix = fuzzed_data_provider.ConsumeIntegral<uint_fast32_t>();
+
+            const std::vector<unsigned char> control = ConsumeRandomLengthByteVector(fuzzed_data_provider);
+            const std::vector<unsigned char> script_bytes = ConsumeRandomLengthByteVector(fuzzed_data_provider);
+
+            rawTapEnv simplicityRawTap;
+            simplicityRawTap.controlBlock = control.data();
+            simplicityRawTap.pathLen = (control.size() - TAPROOT_CONTROL_BASE_SIZE) / TAPROOT_CONTROL_NODE_SIZE;
+            simplicityRawTap.scriptCMR = script_bytes.data();
+            tapEnv* taproot = elements_simplicity_mallocTapEnv(&simplicityRawTap);
```

Comment:
`control` and `script_bytes` are random lengths here, `elements_simplicity_mallocTapEnv` calls `sha256_toMidstate` with `scriptCMR` (script_bytes) as the hash which which has a precondition that this must have a size of 32, which I see in interpreter.cpp is checked 

https://github.com/ElementsProject/elements/blob/simplicity/src/simplicity/sha256.h#L97

Reply:
I can update the documentation for `elements_simplicity_mallocTapEnv`.

Note: the documentation currently says that the control block must be of size `33+pathLen*32`.

Reply:
I suppose technically the control block need only be _at least_ `33+pathLen*32` because for C, if arrays are longer than expected, the extra data is just ignored, which you are doing here.

Reply:
Thanks @roconnor-blockstream, updated